src/server/hset_family.cc

```diff
@@ -248,6 +250,7 @@ OpStatus OpIncrBy(const OpArgs& op_args, string_view key, string_view field, Inc
   }
 
   db_slice.PostUpdate(op_args.db_cntx.db_index, it, key);
+  op_args.shard->search_indices()->AddDoc(key, op_args.db_cntx, pv);
```

Comment:
the same. basically I would like that search_index code would be invisible in the common code if they it's not being used. It's not just about the penalty but also about possible bugs creeping in from the search code into the common flow.

Reply:
We can nullify the search_indices pointer until search is used... WDYT?

Reply:
You could also implement a noop `ShardDocIndices` that does nothing. It would save us the need to check for `nullptr`s upon each usage, but will require adding virtual functions to `ShardDocIndices`

Reply:
But ShardDocIndices is already noop if there are no doc indices. Its a for loop over an empty vector

Reply:
So what would be the benefit of nullifying the pointer? :rofl:

Reply:
I don't know 😄 Maybe both of you just missed that its noop at first glance